src/ViewLayer/RaceModeDisplay/RaceModeDashboardView.cpp

```diff
@@ -187,15 +189,89 @@ void RaceModeDashboardView::initalizeFaultAnimation()
 
 void RaceModeDashboardView::runFaultAnimation()
 {
-    int numberOfActiveFaults = batteryFaultsList_.numberOfActiveFaults() +
-                               motorOneFaultsList_.numberOfActiveFaults() +
-                               motorZeroFaultsList_.numberOfActiveFaults();
+    QVector<QString> batteryFaults = batteryFaultsList_.activeFaultLabels();
+    QVector<QString> motorZeroFaults = motorZeroFaultsList_.activeFaultLabels();
+    QVector<QString> motorOneFaults = motorOneFaultsList_.activeFaultLabels();
 
-    if (numberOfActiveFaults > 0 && faultAnimation_->state() != QAbstractAnimation::Running)
+    if (faultAnimation_->state() != QAbstractAnimation::Running && triggerFaultAnimation(batteryFaults, motorZeroFaults, motorOneFaults))
     {
         faultAnimation_->setDirection(QAbstractAnimation::Forward);
         faultAnimation_->start();
     }
+
+    prevBatteryFaults_ = batteryFaults;
+    prevMotorZeroFaults_ = motorZeroFaults;
+    prevMotorOneFaults_ = motorOneFaults;
+}
+
+bool RaceModeDashboardView::triggerFaultAnimation(QVector<QString>& batteryFaults, QVector<QString>& motorZeroFaults, QVector<QString>& motorOneFaults)
```

Comment:
I think triggerFaultAnimation is a confusing name. It seems more like an _action_, rather than what its actually doing, which is just _checking_ if we should run the fault animation. 

If you agree, then I suggest changing the name (though now that I'm thinking about what to rename the function to, its kinda hard..... - maybe something like **faultAnimationCheck**?).